src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/BatchingWorkQueue.cs

```diff
@@ -182,15 +186,15 @@ private async Task Timer_TickAsync()
 
         private void OnStartingBackgroundWork()
         {
-            if (BlockBackgroundWorkStart != null)
+            if (NotifyBackgroundWorkStarting != null)
             {
-                BlockBackgroundWorkStart.Wait();
-                BlockBackgroundWorkStart.Reset();
+                NotifyBackgroundWorkStarting.Set();
             }
```

Comment:
💭 This whole block can be simplified now right?

```csharp
NotifyBackgroundWorkStarting?.Set();
```

Reply:
Yup!